.pre-commit-config.yaml

```
@@ -16,26 +16,27 @@ repos:
       - id: check-added-large-files
 
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v3.0.3 # v2.7.1
+    rev: ffb6a759a979008c0e6dff86e39f4745a2d9eac4 # v3.1.0
     hooks:
       - id: prettier
         files: \.(md|rst|yml|yaml)
         args: [--prose-wrap=preserve]
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.1.4 # v0.0.261
+    rev: 1d42195ebc67a509991eaf68c2353e0181003a95 # frozen: v0.1.14
+
     hooks:
       - id: ruff
         args: [--fix, --show-fixes, --exit-non-zero-on-fix]
 
   - repo: https://github.com/MarcoGorelli/cython-lint
-    rev: v0.15.0 # v0.15.0
+    rev: 582556b052c898dc484a981bea48de3284a4b8ee # frozen: v0.16.0
     hooks:
       - id: cython-lint
         args: [--no-pycodestyle, --max-line-length=88]
 
   - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.10.1
+    rev: a5d7f70a9792311ac09a74de7eb1feb6fb82d84d # frozen: 24.1.0
```

```suggestion
    rev: 3de533c80a7e95bb6efb213169d269267e9b234b # frozen: 23.12.1
```
revert to previous black version that doesn't reformat for now...